From 13f457673d2fe47af6619f510f0f4851c25a6ebd Mon Sep 17 00:00:00 2001
From: Jason Montleon <jmontleo@redhat.com>
Date: Thu, 19 Mar 2026 12:43:17 -0400
Subject: [PATCH 0002/1239] Revert "net: spacemit: Fix error handling in
 emac_alloc_rx_desc_buffers()"

This reverts commit 7d976a6e1da8259e83852ba202eae2732bbd1909.
---
 drivers/net/ethernet/spacemit/k1_emac.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/spacemit/k1_emac.c b/drivers/net/ethernet/spacemit/k1_emac.c
index 5de69a105168..b49c4708bf9e 100644
--- a/drivers/net/ethernet/spacemit/k1_emac.c
+++ b/drivers/net/ethernet/spacemit/k1_emac.c
@@ -582,9 +582,7 @@ static void emac_alloc_rx_desc_buffers(struct emac_priv *priv)
 						  DMA_FROM_DEVICE);
 		if (dma_mapping_error(&priv->pdev->dev, rx_buf->dma_addr)) {
 			dev_err_ratelimited(&ndev->dev, "Mapping skb failed\n");
-			dev_kfree_skb_any(skb);
-			rx_buf->skb = NULL;
-			break;
+			goto err_free_skb;
 		}
 
 		rx_desc_addr = &((struct emac_desc *)rx_ring->desc_addr)[i];
@@ -609,6 +607,10 @@ static void emac_alloc_rx_desc_buffers(struct emac_priv *priv)
 
 	rx_ring->head = i;
 	return;
+
+err_free_skb:
+	dev_kfree_skb_any(skb);
+	rx_buf->skb = NULL;
 }
 
 /* Returns number of packets received */
-- 
2.54.0

